### PR TITLE
docs: update on hostport for upgrade guide and cni chanining

### PR DESCRIPTION
**Pull request by borkmann**
See commits.

### CODE REVIEW

**Comment by borkmann** [comment]
test-docs-please

**Review by aanm — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

File: Documentation/install/upgrade.rst [hunk 1]
```diff
@@ -415,6 +415,16 @@ IMPORTANT: Changes required before upgrading to 1.8.0
   using Helm ``--set global.cnpStatusUpdates.enabled=true`` or by editing the
   ``ConfigMap`` directly.
 
+* Prior to 1.8 release, Cilium's eBPF-based kube-proxy replacement was not able
+  to handle Kubernetes HostPort feature and therefore CNI chaining with the
+  ``portmap`` plugin (``global.cni.chainingMode=portmap``) was necessary while
+  turning off the kube-proxy replacement (``global.kubeProxyReplacement=disabled``).
+  Starting from 1.8, CNI chaining is no longer necessary, meaning Cilium can be
+  used natively to handle HostPort when running with Cilium's kube-proxy replacement.
+  That is, for ``global.kubeProxyReplacement=probe`` and ``global.kubeProxyReplacement=strict``
```
Comment (brb) [inline Documentation/install/upgrade.rst:424]: Maybe we should mention that in the case of `probe`, it will be enabled if BPF NodePort was enabled?

Comment (borkmann) [reply Documentation/install/upgrade.rst:424]: Good point, yes. Adding.

**Comment by borkmann** [comment]
test-docs-please

**Review by brb — APPROVED** [review]

**Pull request merged**